test/chunk_info.c

```diff
@@ -544,14 +544,15 @@ test_get_chunk_info_highest_v18(hid_t fapl)
     /* Allocate input (compressed) buffer */
     inbuf = HDmalloc(z_dst_nbytes);
 
-    /* Set chunk size to the compressed chunk size and the chunk point
-       to the compressed data chunk */
-    chunk_size = (hsize_t)z_dst_nbytes;
```

Comment:
Setting chunk_size here sets it to the size of the allocated buffer and writes the entire buffer, including any space after the compressed region, which is uninitialized.